### ISSUE TITLE
Getting original publish date from git history?

### ISSUE THREAD

**Issue by onnimonni** [gohugoio/hugo#11810]
Hugo already has the the ability to get modified date from git but could we also get the original creation date from git with the:
```
enableGitInfo = true
```

This should be possible with git:
```
$ git log --diff-filter=A --follow --format=%aD -1 -- <file-name-here>
```

**Comment by bep** [comment]
We already have `CommitDate` and  `AuthorDate` using these flags:
```
%ci
committer date, ISO 8601-like format
%ai
author date, ISO 8601-like format
```

I have looked in the Git docs and found this:

```
%ad
author date (format respects --date= option)
```

What is the difference between the two author dates?

**Comment by onnimonni** [comment]
Can you point me where the `CommitDate` is documented? It's very likely I just wasn't aware of this 🙇

**Comment by bep** [comment]
Yea, commitDate was missing from the documentation (I have updated it), but this is the struct we're returning:

```go

// GitInfo provides information about a version controlled source file.
type GitInfo struct {
	// Commit hash.
	Hash string `json:"hash"`
	// Abbreviated commit hash.
	AbbreviatedHash string `json:"abbreviatedHash"`
	// The commit message's subject/title line.
	Subject string `json:"subject"`
	// The author name, respecting .mailmap.
	AuthorName string `json:"authorName"`
	// The author email address, respecting .mailmap.
	AuthorEmail string `json:"authorEmail"`
	// The author date.
	AuthorDate time.Time `json:"authorDate"`
	// The commit date.
	CommitDate time.Time `json:"commitDate"`
}
```

**Comment by gmantele** [comment]
_[By looking at all issues, it seems this one matches my question (if not, sorry for adding noise to this discussion ; just tell me and I'll open another one).]_

Currently `AuthorDate` and `CommitDate` are set to the same value: the last commit about the file. It is indeed a really awesome feature: being able to add an update date on our final web page to show when it has been modified. As the original author of this Issue, I would also be interested in the creation date of a committed file (if I understood well, it should be `AuthorDate`).

It is then still normal that both dates are set to the same one?
It seems the resolution of this Issue has been postponed to v0.125.0. I assume we have to wait for this version of Hugo to have a clear difference between `AuthorDate` and `CommitDate`, isn't it?

**Comment by gmantele** [comment]
Ok. I better understand now. Thank you very much.

Indeed, 56 seconds is very slow. Slowing down a so fast site generation would really be a shame. So, forget about my question/request. And sorry for having spoiled this thread.

**Comment by Finkregh** [comment]
That feature is still relevant, not having to maintain the date in the metadata

**Comment by jmooring** [comment]
It may be relevant but it is very expensive. See my previous comment.

**Comment by Finkregh** [comment]
Yep, the user would have to explicitly enable that feature I guess :)
